keps/sig-architecture/2527-clarify-status-observations-vs-rbac/README.md

```diff
@@ -0,0 +1,697 @@
+and apply similar rules to `spec` and `status` with regards to writes (can't
+write to `control` through the main resource, can't write to `spec` or `status`
+through the `control` subresource).  We would also need to add it to CRDs as an
+available subresource.
+
+#### Examples
+
+1) User creates a Service and sets `Service.spec.type = LoadBalancer`.  The
+cloud controller sets `Service.control.loadBalancer.ingress.ip` in response.  If
+the user sets `Service.spec.loadBalancerIP` to a specific value, the cloud
+controller will either successfully use that IP and and set it as before (ACK),
+or it will not set any value in `Service.control.loadBalancer.ingress` (NAK).
+
+2) Given a Pod, user patches `Pod.spec.containers[0].resources.requests[cpu]`
+to a new value.  Kubelet sees this as a request and, if viable, sets
+`Pod.control.containerStatuses[0].resources.requests[cpu]` to the same value.
+
+3) User creates a `Bucket`, setting `Bucket.spec.instance = artifacts-prod`.
+The bucket controller verifies that the namespace is allowed to use the
+artifacts-prod bucket and, if so, sets `Bucket.control.instance` to the same
+value.
+
+#### Tradeoffs
+
+Pro: Clarifies the meaning of status.
+
+Pro: Possibly clarifies the roles acting on a resource.
+
+Con: Requires a lot of implentation and possibly steps on existing uses of the
+field name.
+
+Con: Net-new concept requires new documentation and socialization.
+
+Con: Incompatible with existing uses of status for this.
+
+### Option 3: Sub-divide spec
+
+Keep and strengthen the idea that status fields must be from observation.
+Segregate controller-owned fields to a sub-stanza of spec.  Create a new
+access-control mechanism (or extend RBAC) to provide field-by-field access.
+
+#### Examples
+
+1) User creates a Service and sets `Service.spec.type = LoadBalancer`.  The
+cloud controller sets `Service.spec.control.loadBalancer.ingress.ip` in
+response.  If the user sets `Service.spec.loadBalancerIP` to a specific value,
+the cloud controller will either successfully use that IP and and set it as
+before (ACK), or it will not set any value in
+`Service.spec.control.loadBalancer.ingress` (NAK).
+
+2) Given a Pod, user patches `Pod.spec.containers[0].resources.requests[cpu]`
+to a new value.  Kubelet sees this as a request and, if viable, sets
+`Pod.spec.control.containerStatuses[0].resources.requests[cpu]` to the same
+value.
+
+3) User creates a `Bucket`, setting `Bucket.spec.instance = artifacts-prod`.
+The bucket controller verifies that the namespace is allowed to use the
+artifacts-prod bucket and, if so, sets `Bucket.spec.control.instance` to the same
+value.
+
+#### Tradeoffs
+
+Pro: Retains purity of status.
+
+Con: Confuses the meaning of spec.
+
+Con: Can collide with existing uses of the field name.
+
+Con: Needs a whole new access model.
+
+Con: Likely needs a new subresource.
+
+#### Notes
+
+This model is included for completeness.  I do not expect ANYONE to endorse it.
```

Comment:
I was literally about to ask if options 2 and 3 were here to make option 1 seem amazing :)